script.js

```diff
@@ -1,4 +1,421 @@
+var ROLL_PLAYER_TWO = "ROLL_PLAYER_TWO";
+var ORDER_SELECTION_PLAYER_TWO = "ORDER_SELECTION_PLAYER_TWO";
+
+// Creating Game State Variables
+var gameState = PLAYERS;
+var currentRolls = [];
+var currentValue;
+var currentPlayer = 1;
+var playerOne;
+var playerTwo;
+var knockoutScore = [0, 0];
+
+// Helper Function to Set Number of Players
+var setNumPlayers = function (number) {
+  if (Number.isNaN(Number(number))) {
+    var outputValue = "Invalid Entry. Please key in an integer.";
+    return outputValue;
+  } else {
+    number = Number(number);
+    numPlayers = number;
+    var outputValue = `You have selected to play with ${numPlayers} players.`;
+    gameState = DICE;
+    return outputValue;
+  }
+};
+
+// Helper Function for Number of Dice
+var setNumDice = function (number) {
+  if (Number.isNaN(Number(number))) {
+    var outputValue = "Invalid Entry. Please key in an integer.";
+    return outputValue;
+  } else {
+    number = Number(number);
+    numDice = number;
+    var outputValue = `You have selected to play with ${numDice} dice.`;
+    gameState = LOW_HIGH;
+    return outputValue;
+  }
+};
+
+// Helper Function for Single Dice Roll
+var singleDiceRoll = function () {
+  var outputValue = Math.floor(Math.random() * 6) + 1;
+  console.log("Dice Roll:" + outputValue);
+  return outputValue;
+};
+
+// Helper Function for Multiple Dice Roll. Returns an Array of Dice Rolls.
+var multipleDiceRoll = function (number) {
+  if (Number.isNaN(Number(number))) {
+    var outputValue = "Invalid Entry. Please key in an integer.";
+    return outputValue;
+  } else {
+    var outputArray = [];
+    for (var counter = 0; counter < number; counter += 1) {
+      outputArray.push(singleDiceRoll());
+    }
+    console.log("Dice Array:" + outputArray);
+    return outputArray;
+  }
+};
+
+// Helper Function to Select Game Mode
+var setGameMode = function (input) {
+  if (input == KNOCKOUT || input == ARCADE) {
+    gameMode = input;
+    outputValue = `You have selected the ${gameMode} game mode. <br>`;
+    if (input == ARCADE) {
+      generateScoreboard();
+      gameState = ROLL;
+      outputValue += `Player ${currentPlayer} will roll first.`;
+    } else {
+      generateFixtures();
+      gameState = ROLL_PLAYER_ONE;
+      outputValue += `The first fixture is Player ${playerOne} VS Player ${playerTwo}.<br>Player ${playerOne} will roll first.`;
+    }
+    return outputValue;
+  } else {
+    outputValue = `Invalid entry. Please input either ${KNOCKOUT} or ${ARCADE}.`;
+    return outputValue;
+  }
+};
+
+// Helper Function to Select Low/High
+var setLowHigh = function (input) {
+  if (input == LOW || input == HIGH) {
+    lowHigh = input;
+    outputValue = `You have selected that the ${lowHigh}EST roll wins.`;
+    gameState = AUTO_CHOOSE;
+    return outputValue;
+  } else {
+    outputValue = `Invalid entry. Please input either ${LOW} or ${HIGH}.`;
+    return outputValue;
+  }
+};
```

Comment:
I like the use of helper functions and comments to make your code much more organized and readable